### ISSUE TITLE
Can't `vagrant up` a Hyper-v windows 10 box built with packer due to the `AutomaticCheckpointsEnabled` setting

### ISSUE THREAD

**Issue by MiguelAlho** [hashicorp/vagrant#10175]
### Vagrant version
2.1.4

### Host operating system
Windows 10 Enterprise 14393

### Guest operating system
Windows 10 Pro N Build 17134 (occurs with or without updates added through Packer, using the `en_windows_10_business_editions_version_1803_updated_march_2018_x64_dvd_12063333.iso`) 

### Vagrantfile
```ruby
Vagrant.configure("2") do |config|
    config.vm.box = "BaseImage"
    config.vm.box_url = "//remotepathtobox/windows_10_base_hyperv_0.0.2.box"

    config.vm.network "private_network", bridge: "externalSwitch"

   config.vm.provider "hyperv" do |h|
      h.ip_address_timeout = 600
      h.enable_virtualization_extensions = true
      h.enable_automatic_checkpoints = false
      h.linked_clone = true
    end
```

It occurs with or without the config.vm.provider definition

### Debug output
https://gist.github.com/MiguelAlho/a3d4335b35add0320c20c05f49542ada

### Expected behavior
VM should start 

### Actual behavior
Box can be added to vagrant but VM fails to start with the following error message when `vagrant up` is called:

```
 Bringing machine 'default' up with 'hyperv' provider...
==> default: Verifying Hyper-V is enabled...
==> default: Verifying Hyper-V is accessible...
    default: Configuring the VM...
An error occurred while executing a PowerShell script. This error
is shown below. Please read the error message and see if this is
a configuration error with your system. If it is not, then please
report a bug.

Script: configure_vm.ps1
Error:

Failed to disable automatic checkpoints on VM: A parameter cannot be found that matches parameter name 'AutomaticCheckpointsEnabled'.
```

### Steps to reproduce
1. `vagrant up`

### also:
- I only started to see this behaviour after upgrading both vagrant (and packer) to 2.1.4 (and packer at 2.1.5)
- With the box visible in the Hyper-V manager, I can boot up the VM manually
- If I disable checkpoints in the VM settings in Hyper-V manager (manually through the GUI), when I call `vagrant up`, vagrant still fails to boot the machine and the chackpoint setting ends up re-enabled.
- I've built the box with packer both with and without windows updates applied (same behaviour with either one) and with or without the `enable_automatic_checkpoints` explicitly set to false in the vagrant file I have in the packer project repo.

```
  #... 
  #for hyperv a longer timeout is necessary, especially for first boot
  config.vm.provider "hyperv" do |h|
    h.ip_address_timeout = 600
    h.enable_virtualization_extensions = true
    h.enable_automatic_checkpoints = false
    #h.differencing_disk = true
  end
```

### References
Are there any other GitHub issues (open or closed) that should be linked here?
For example:
- https://github.com/hashicorp/packer/issues/5371 (from the packer repo)
- https://github.com/hashicorp/packer/pull/5374 (the PR)
- https://github.com/hashicorp/vagrant/pull/9999 (vagrant PR, possibly related)
- https://github.com/hashicorp/vagrant/issues/9977 (issue for the previous PR)
- https://github.com/hashicorp/vagrant/issues/8988 (similar / related)

**Comment by MiguelAlho** [comment]
Sorry, I forgot to include the context: 

I'm using Packer to build a Win 10 Hyper-V vm. I expect to use this as a base image to build tool specific images to use in build and test processes, where I can't use Docker at the moment. My current expectation is that I can:
- build the base box using packer; (scripted build I can put on the CI server)
- publish the box to a remote share (for a private repo)
- use vagrant in tool specific CI builds to create a copy of the image with the tool and it's prerequisites installed.
- publish this modified / augmented box to the share (with a new name and version) so that i can be uped in other project's CI builds

The first step seems to be running correctly (I can build the VM and publish it), but I'm failing at the second step. I'm a bit of a noob with provisioning and Vagrant, so I may be going about this wrong, for which, if so, I'd apreciate the heads-up.

**Comment by johnmccain** [comment]
I am experiencing the same issue using a fresh install of Vagrant version 2.1.5 on Windows 10 Version 10.0.14393.

Debug output:
https://gist.github.com/johnmccain/592b5c7147c177bd011f5597e5901e64

**Comment by i-nikolov** [comment]
I am experiencing the same issue. Vagrant 2.1.5

**Comment by gerardog** [comment]
Workaround:
Edit `C:\HashiCorp\Vagrant\embedded\gems\2.1.5\gems\vagrant-2.1.5\plugins\providers\hyperv\scripts\configure_vm.ps1`

Remove:

```
    if($autochecks -eq 1 -and (Get-Command Hyper-V\Set-VM).Parameters["AutomaticCheckpointsEnabled"] -eq $null) {
        Write-ErrorMessage "AutomaticCheckpointsEnabled is not available"
        exit 1
    }
     Hyper-V\Set-VM -VM $VM -AutomaticCheckpointsEnabled $autochecks
```

**Comment by chrisroberts** [comment]
This was fixed in #10265 and was released in 2.2.0

**Comment by gerardog** [comment]
Yes, thank you. But 2.2.0 and 2.2.1 are not working for me (windows 10+hyper-v+windows server image) because of #10438.
